### ISSUE TITLE
Improve MercatorTiledImageLayer

### ISSUE THREAD

**Issue by EMaksymenko** [WorldWindEarth/worldwindjs#29]
MercatorTiledImageLayer can be improved by translating following TS to JS:
```
export abstract class MercatorTiledImageLayer extends WorldWind.MercatorTiledImageLayer {

  private static readonly FULL_SPHERE = 360;

  private readonly firstLevelOffset;

  private static gudermannian(percent: number): number {
    return Math.atan(Math.sinh(percent * Math.PI)) * 180 / Math.PI;
  }

  private static levelZeroDelta(firstLevelOffset: number): WorldWind.Location {
    const levelZeroDelta = MercatorTiledImageLayer.FULL_SPHERE / (1 << firstLevelOffset);
    return new WorldWind.Location(levelZeroDelta / 2, levelZeroDelta);
  }

  protected constructor(name: string, numLevels: number, firstLevelOffset: number, tileSize: number, overlay: boolean, imageFormat: string){
    super(new WorldWind.Sector(
        MercatorTiledImageLayer.gudermannian(-1), MercatorTiledImageLayer.gudermannian(1),
        - MercatorTiledImageLayer.FULL_SPHERE / 2, MercatorTiledImageLayer.FULL_SPHERE / 2
      ), MercatorTiledImageLayer.levelZeroDelta(firstLevelOffset), numLevels - firstLevelOffset,
      imageFormat, name, tileSize, tileSize);
    this.firstLevelOffset = firstLevelOffset;
    (this as WorldWind.Layer).pickEnabled = false;
    (this as WorldWind.Layer).displayName = name;
    (this as WorldWind.MercatorTiledImageLayer).detectBlankImages = overlay;
    (this as WorldWind.MercatorTiledImageLayer).imageSize = tileSize;
  }

  protected mapSizeForLevel(levelNumber: number): number {
    return (this as WorldWind.MercatorTiledImageLayer).imageSize << (levelNumber + this.firstLevelOffset);
  }

  protected resourceUrlForTile(tile: WorldWind.ImageTile, imageFormat: string): string {
    return this.getImageSourceUrl(tile.column, tile.row, tile.level.levelNumber + this.firstLevelOffset);
  }

  protected abstract getImageSourceUrl(x: number, y: number, z: number): string;

}
```
This code makes Mercator layers processing the same as in Android and Java.

**Comment by EMaksymenko** [comment]
Implemented by PR #40. Please review.

**Comment by EMaksymenko** [comment]
Now it is possible to create online maps like:

OpenStreetMap:
```
import {MercatorTiledImageLayer} from '../MercatorTiledImageLayer';

export class OSMLayer extends MercatorTiledImageLayer {

  public static readonly NAME = 'OpenStreetMap';

  public constructor() {
    super(OSMLayer.NAME, 19, 'image/png', false, OSMLayer.NAME, 256, 3);
  }

  protected getImageSourceUrl(x: number, y: number, z: number): string {
    const abc = 'abc'.charAt(Math.round(Math.random() * 2));
    return 'https://' + abc + '.tile.openstreetmap.org/' + z + '/' + x + '/' + y + '.png';
  }

}
```
OpenTopoMap:
```
import {MercatorTiledImageLayer} from '../MercatorTiledImageLayer';

export class OTMLayer extends MercatorTiledImageLayer {

  public static readonly NAME = 'OpenTopoMap';

  public constructor() {
    super(OTMLayer.NAME, 17, 'image/png', false, OTMLayer.NAME, 256, 3);
  }

  protected getImageSourceUrl(x: number, y: number, z: number): string {
    const abc = 'abc'.charAt(Math.round(Math.random() * 2));
    return 'https://' + abc + '.tile.opentopomap.org/' + z + '/' + x + '/' + y + '.png';
  }

}

```

Wiki:
```
import {MercatorTiledImageLayer} from '../MercatorTiledImageLayer';

export enum WikiLayerType {MAP = 'map', HYBRID = 'hybrid'}

export class WikiLayer extends MercatorTiledImageLayer {

  public static readonly NAME = 'Wiki';

  public constructor(private readonly type: WikiLayerType = WikiLayerType.HYBRID) {
    super(WikiLayer.NAME + type, 19, WikiLayerType.HYBRID === type ? 'image/png' : 'image/jpg', WikiLayerType.HYBRID === type, WikiLayer.NAME + type, 256, 3);
  }

  protected getImageSourceUrl(x: number, y: number, z: number): string {
    const i = x % 4 + (y % 4) * 4;
    return 'http://i' + i + '.wikimapia.org/?lng=1&x=' + x + '&y=' + y + '&zoom=' + z + '&type=' + this.type;
  }

}
```

Google:
```
import {MercatorTiledImageLayer} from '../MercatorTiledImageLayer';

export enum GoogleLayerType {
  ROADMAP = 'GoogleRoadMap|m|false|image/png',
  // ROADMAP2 = 'GoogleRoadMap2|r|false|image/png',
  TERRAIN = 'GoogleTerrainMap|p|false|image/png',
  // TERRAIN_ONLY = 'GoogleTerrainOnly|t|false|image/png',
  // HYBRID = 'GoogleHybrid|y|false|image/png',
  SATELLITE = 'GoogleSatellite|s|false|image/jpg',
  ROADS = 'GoogleRoads|h|true|image/png',
  TRAFFIC = 'GoogleTraffic|h,traffic&style=15|true|image/png'
}

export class GoogleLayer extends MercatorTiledImageLayer {

  private static type: GoogleLayerType;
  private static typeArray: string[];

  private readonly lyrs: string;

  private static splitType(type: GoogleLayerType): string[] {
    // Split enum parameters to values array once
    if (!GoogleLayer.typeArray || GoogleLayer.type !== type) {
      GoogleLayer.typeArray = type.split('|');
    }
    return GoogleLayer.typeArray;
  }

  public constructor(type: GoogleLayerType) {
    super(GoogleLayer.splitType(type)[0], 22, GoogleLayer.splitType(type)[3], GoogleLayer.splitType(type)[2] === 'true', GoogleLayer.splitType(type)[0], 256, 3);
    this.lyrs = GoogleLayer.splitType(type)[1];
  }

  protected getImageSourceUrl(x: number, y: number, z: number): string {
    const language = window.navigator.language.split('-')[0];
    return 'https://mt.google.com/vt/lyrs=' + this.lyrs + '&x=' + x + '&y=' + y + '&z=' + z + '&hl=' + language;
  }

}
```
